src/content/docs/en/reference/loader-reference.mdx

```diff
@@ -3,6 +3,324 @@ title: Astro Loader API
+      logger.info("Loading posts");
+      const feed = loadFeed(feedUrl);
+      store.clear();
+
+      for (const item of feed.items) {
+        const data = await parseData({
+          id: item.guid,
+          data: item,
+        });
+
+        const digest = generateDigest(data);
+
+        store.set({
+          id,
+          data,
+          digest,
+        });
+      }
+    },
+  };
+}
+```
+
+#### `watcher`
+**Type**: `FSWatcher`
+
+When running in dev mode, this is a filesystem watcher that can be used to trigger updates. See [`ViteDevServer`](https://vitejs.dev/guide/api-javascript.html#vitedevserver) for more information.
+
+```ts title="Extract from the file() loader" {8-13}
+return {
+  name: 'file-loader',
+  load: async ({ config, store, watcher }) => {
+    const url = new URL(fileName, config.root);
+    const filePath = fileURLToPath(url);
+    await syncData(filePath, store);
+
+    watcher?.on('change', async (changedPath) => {
+      if (changedPath === filePath) {
+        logger.info(`Reloading data from ${fileName}`);
+        await syncData(filePath, store);
+      }
+    });
+  },
+};
+```
+#### `refreshContextData`
+**Type**: `Record<string, unknown>`
+
+If the loader has been triggered by an integration, this may optionally contain extra data set by that integration. It is only set when the loader is triggered by an integration. See the [`astro:server:setup`](/en/reference/integrations-reference/#astroserversetup) hook reference for more information.
+
+```ts title=loader.ts {5-8}
+export function myLoader(options: { url: string }): Loader {
+  return {
+    name: "my-loader",
+    load: async ({ refreshContextData, store, logger }) => {
+      if(refreshContextData?.webhookBody) {
+        logger.info("Webhook triggered with body");
+        processWebhook(store, refreshContextData.webhookBody);
+      }
+      // ...
+    },
+  };
+}
+```
+
+### `DataStore`
+The data store is a loader's interface to the content collection data. It is a key value store, scoped to the collection, so a loader can only access the data for its own collection.
+
+#### `get`
+**Type**: `(key: string) => DataEntry | undefined`
+
+Get an entry from the store by its ID. Returns `undefined` if the entry does not exist. 
+
+```ts
+const existingEntry = store.get("my-entry");
+```
+
+The returned object has the following properties:
+```ts
+interface DataEntry<TData extends Record<string, unknown>> {
+	/** The ID of the entry. Unique per collection. */
+	id: string;
+	/** The parsed entry data */
+	data: TData;
+	/** The file path of the content, if applicable. Relative to the site root. */
+	filePath?: string;
+	/** The raw body of the content, if applicable. */
+	body?: string;
+	/** An optional content digest, to check if the content has changed. */
+	digest?: number | string;
+	/** The rendered content of the entry, if applicable. */
+	rendered?: RenderedContent;
+}
+```
+
+#### `set`
+
+**Type**: `(opts: Opts) => boolean`
```

Comment:
Nitpicking for the formatting.
Otherwise, does this property refer to [`MutableDataStore`](https://github.com/withastro/astro/blob/1c64ae304d3c008a776e98e48fd3ece8be7b1fb5/packages/astro/src/content/mutable-data-store.ts#L16C14-L16C30) because I can't find it in [`DataStore`](https://github.com/withastro/astro/blob/1c64ae304d3c008a776e98e48fd3ece8be7b1fb5/packages/astro/src/content/data-store.ts#L44). If so, I think the type should be updated to be clearer for readers:
```suggestion
<p>

**Type**: `(collectionName: string, key: string, value: unknown) => boolean`
</p>
```

Reply:
No, this has changed in the 5.0 branch. The `DataStore` interface is what was previous called `ScopedDataStore`. The previous `DataStore` is renamed to `ImmutableDataStore`.

See https://github.com/withastro/astro/blob/next/packages/astro/src/content/mutable-data-store.ts#L333